### PR TITLE
Jeffmausolf/6968 update python version and tool extensions

### PR DESCRIPTION
**Pull request by mausolfj**
Updated python version per recommendation of Data Science Lead then reworked devcontainer to align with new schema changes

### CODE REVIEW

**Comment by mausolfj** [comment]
Updated python version in Dockerfile and multiple pyenv.cfg files then updated format of devcontainer to align with new schema.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -23,58 +23,61 @@
...
-            "**/.git/subtree-cache/**": true,
-            "**/node_modules/*/**": true,
-            "**/.python_packages/*/**": true
-          },
-		"files.associations": {
-			"*.workbook": "[jsonc]"
+	// Set *default* container specific settings values on container create.
+	"customizations": {
+    	"vscode": {
+			"settings": {
```
Comment (georearl) [inline .devcontainer/devcontainer.json:29]: could we indent correctly

Comment (georearl) [inline .devcontainer/devcontainer.json:29]: could we adjust indenting

**Review by georearl — DISMISSED** [review]

**Review by ryonsteele — APPROVED** [review]

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -23,58 +23,59 @@
...
-            "**/.git/subtree-cache/**": true,
-            "**/node_modules/*/**": true,
-            "**/.python_packages/*/**": true
-          },
-		"files.associations": {
-			"*.workbook": "[jsonc]"
+	// Set *default* container specific settings values on container create.
+	"customizations": {
+		"vscode": {
+			"settings": {
```
Comment (dayland) [inline .devcontainer/devcontainer.json:29]: I am still getting the following warnings after rebuilding my devconainer with these changes... 
image

```
2024-03-01 13:11:06.262 [info] Experiment 'pythonRecommendTensorboardExt' is active
2024-03-01 13:11:06.262 [info] Experiment 'pythonREPLSmartSend' is active
2024-03-01 13:11:06.262 [info] Experiment 'pythonSurveyNotification' is active
2024-03-01 13:11:06.262 [info] Experiment 'pythonTerminalEnvVarActivation' is active
2024-03-01 13:11:06.262 [info] Experiment 'pythonTestAdapter' is active
2024-03-01 13:11:06.262 [info] Default formatter is set to null for workspace /workspaces/info-asst
2024-03-01 13:11:06.262 [error] Following setting is deprecated: "python.linting.pylintEnabled"
2024-03-01 13:11:06.262 [error] All settings starting with "python.linting." are deprecated and can be removed from settings.
2024-03-01 13:11:06.262 [error] Linting features have been moved to separate linter extensions.
2024-03-01 13:11:06.262 [error] See here for more information: https://code.visualstudio.com/docs/python/linting
2024-03-01 13:11:06.262 [error] Please install "pylint" extension: https://marketplace.visualstudio.com/items?itemName=ms-python.pylint
2024-03-01 13:11:06.262 [info] Test server listening.
2024-03-01 13:11:06.449 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.484 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.484 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.484 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.484 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.484 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.484 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.484 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.484 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.484 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.534 [info] > conda info --json
2024-03-01 13:11:06.549 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.549 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.549 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
2024-03-01 13:11:06.586 [info] > /usr/bin/python3 -I ~/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/interpreterInfo.py
2024-03-01 13:11:06.610 [info] > /usr/local/bin/python -I ~/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/interpreterInfo.py
2024-03-01 13:11:06.643 [info] > ./.venv/bin/python -I ~/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/get_output_via_markers.py ~/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/interpreterInfo.py
2024-03-01 13:11:06.678 [info] Python interpreter path: ./.venv/bin/python
2024-03-01 13:11:06.794 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2024-03-01 13:11:06.794 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2024-03-01 13:11:06.911 [info] > . ./.venv/bin/activate && echo 'e8b39361-0157-4923-80e1-22d70d46dee6' && python ~/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/printEnvVariables.py
2024-03-01 13:11:06.911 [info] shell: bash
2024-03-01 13:11:06.913 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2024-03-01 13:11:06.913 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2024-03-01 13:11:06.999 [info] > . ./.venv/bin/activate && echo 'e8b39361-0157-4923-80e1-22d70d46dee6' && python ~/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/printEnvVariables.py
2024-03-01 13:11:06.999 [info] shell: bash
2024-03-01 13:11:07.007 [warning] Retry failed, storage update failed for key PYTHON_ENV_INFO_CACHE
2024-03-01 13:11:07.049 [info] > /usr/bin/python3 ~/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/printEnvVariables.py
2024-03-01 13:11:07.049 [info] shell: bash
2024-03-01 13:11:07.070 [info] Send text to terminal: /usr/bin/python3 /home/vscode/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/printEnvVariablesToFile.py /home/vscode/.vscode-server/extensions/ms-python.python-2024.2.1/pythonFiles/deactivate/bash/envVars.txt
2024-03-01 13:11:08.030 [info] Starting Pylance language server.
```

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM mcr.microsoft.com/vscode/devcontainers/python:0-3.10-bullseye
+FROM mcr.microsoft.com/vscode/devcontainers/python:3.10-bullseye
```
Comment (dayland) [inline .devcontainer/Dockerfile:1]: Python is now up to 3.12 available, Is there a reason we chose to stay with 3.10?

Comment (ryonsteele) [reply .devcontainer/Dockerfile:1]: Good point. I do not see a 3.12 official MS container for that flavor and 3.12 But I do see one for 3.11
https://[mcr.microsoft.com/v2/vscode/devcontainers/python/tags/list](https://mcr.microsoft.com/v2/vscode/devcontainers/python/tags/list)

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (ryonsteele) [reply .devcontainer/devcontainer.json:29]: Partly my fault, I told him I didn't think we needed to update the settings. But i thought the ask was just python and extensions checking. Sorry Jeff

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (mausolfj) [reply .devcontainer/Dockerfile:1]: Yes, I originally tried with 3.12 but was overwhelmed with errors. 3.11 was much better but then I talked with our Data Science Lead who recommended 3.10.12.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (mausolfj) [reply .devcontainer/devcontainer.json:29]: NP Ryon, I'll test locally with rebuild without cache to duplicate this issue. Thanks for bringing this to my attention Daylan

File: .devcontainer/Dockerfile (hunk 3 above)
Comment (dayland) [reply .devcontainer/Dockerfile:1]: ok.. sounds good.

**Review by dayland — APPROVED** [review]

**Review by dayland — APPROVED** [review]

**Pull request merged**